src/progress_bar.rs

```diff
@@ -691,20 +691,8 @@ impl MultiProgress {
                 }
             }
 
-            // Split orphan lines out of the draw state, if any
-            let lines = if draw_state.orphan_lines > 0 {
-                let split = draw_state.lines.split_at(draw_state.orphan_lines);
-                orphan_lines.extend_from_slice(split.0);
-                split.1.to_vec()
-            } else {
-                draw_state.lines
-            };
-
-            let draw_state = ProgressDrawState {
-                lines,
-                orphan_lines: 0,
-                ..draw_state
-            };
+            let orpan = draw_state.split_off_orphan_lines();
```

Comment:
By default, I'm not a fan of adding single-use functions or methods. I find that the added indirection can often make it harder to understand what's going on. What is the point of this change?